File: MatrixEngine/Engine/ECS/Entity.cpp
#include "Entity.h"
using me::Entity;
me::Entity::Entity()
{
	id = counter++;
}

namespace me
{
	bool operator<(const Entity& a, const Entity& b)
	{
		return a.id < b.id;
	}
	unsigned long long Entity::counter = 0;

}

File: src/entity.rs
use std::sync::atomic::AtomicUsize;

static COUNTER: AtomicUsize = AtomicUsize::new(0);

#[derive(Clone, Copy, PartialEq, Eq, Hash,Debug)]
pub struct Entity(usize);

impl Default for Entity {
    fn default() -> Self {
        Self(COUNTER.fetch_add(1, std::sync::atomic::Ordering::Relaxed))
    }
}
